tapdb/sqlc/queries/assets.sql

```diff
@@ -1016,3 +1017,22 @@ FROM genesis_assets assets
 JOIN assets_meta
     ON assets.meta_data_id = assets_meta.meta_id
 WHERE assets.asset_id = $1;
+
+-- Upsert a record into the mint_anchor_uni_commitments table.
+-- If a record with the same batch_id and group_key already exists, update the
+-- existing record. Otherwise, insert a new record.
+-- name: UpsertMintAnchorUniCommitment :one
+INSERT INTO mint_anchor_uni_commitments (
+    id, batch_id, tx_output_index, taproot_internal_key, group_key
+)
+VALUES ($1, $2, $3, $4, $5)
+ON CONFLICT(batch_id, group_key) DO UPDATE SET
```

Comment:
There is no unique key that spans over these two fields, so this will never trigger.
I wonder why SQLite or sqlc didn't pick that up?
We'll need an explicit unique key if we want this combination to be unique.
Also makes me think we should have a unit test that covers the specific case.

Reply:
I think that I'm going to pair on `(batch_id, tx_output_index)` instead with `UNIQUE` and `CONFLICT`. Which effectively means: that each batch could have multiple different pre-commitment outputs and each pre-commitment output would relate to a single asset group.